projects/sbb-esta/angular-business/chip/src/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,260 @@
+  @Input()
+  get required(): boolean {
+    return this._required;
+  }
+  set required(value: boolean) {
+    this._required = coerceBooleanProperty(value);
+    this.stateChanges.next();
+  }
+
+  /** @docs-private */
+  @HostBinding('class.sbb-chip-input-active')
+  get _isActive() {
+    return !this.disabled && this.focused;
+  }
+
+  @ViewChild('chipInputTextfield', { static: false })
+  inputElement: ElementRef;
+
+  /** Value of the chip input control. */
+  get value(): any {
+    return this._value;
+  }
+  set value(newValue: any) {
+    if (newValue !== this._value) {
+      this._value = newValue;
+    }
+  }
+  private _value: string[] = [];
+
+  /**
+   * Implemented as part of FormFieldControl.
+   * @docs-private
+   */
+  get inputId() {
+    return this.id;
+  }
+
+  /** Whether the chip-input has a value. */
+  get empty(): boolean {
+    return !this.selectionModel || this.selectionModel.isEmpty();
+  }
+
+  /**
+   * Event that emits whenever the raw value of the chip-input changes. This is here primarily
+   * to facilitate the two-way binding for the `value` input.
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<any> = new EventEmitter<any>();
+
+  focused = false;
+  errorState: boolean;
+  inputModel = '';
+  origin = new AutocompleteOriginDirective(this._elementRef);
+  selectionModel: SelectionModel<string>;
+
+  private _onTouchedCallback: () => void;
+  private _onChangeCallback: (_: any) => void;
+
+  /** Whether filling out the chip-input is required in the form. */
+  private _required = false;
+
+  /** The aria-describedby attribute on the chip-input for improved a11y. */
+  private _ariaDescribedby: string;
+
+  /** Unique id for this input. */
+  private _uid = `sbb-chip-input-${nextUniqueId++}`;
+
+  /** Emits when the state of the option changes and any parents have to be notified. */
+  readonly stateChanges = new Subject<void>();
+
+  constructor(
+    @Self() @Optional() public ngControl: NgControl,
+    private _elementRef: ElementRef,
+    private _changeDetectorRef: ChangeDetectorRef
+  ) {
+    if (this.ngControl) {
+      // Note: we provide the value accessor through here, instead of
+      // the `providers` to avoid running into a circular import.
+      this.ngControl.valueAccessor = this;
+    }
+
+    // Force setter to be called in case id was not specified.
+    this.id = this.id;
+  }
+
+  ngOnInit(): void {
+    this.selectionModel = new SelectionModel<string>(true);
+    this._initializeSelection();
+    if (this.autocomplete) {
+      this.autocomplete.optionSelected.subscribe(event => this.onSelect(event.option.value));
+    }
+  }
+
+  /**
+   * Sets the chip input's value. Part of the ControlValueAccessor interface
+   * required to integrate with Angular's core forms API.
+   *
+   * @param value New value to be written to the model.
+   */
+  writeValue(value: string[]): void {
+    if (this.selectionModel) {
+      value.forEach(v => this.onSelect(v));
```

Comment:
This would trigger change detection and an emit for every single value
It might be better to only do that after all values are assigned
Maybe compare with `writeValue` of the select component